platform-hub-api/app/controllers/me_controller.rb

```diff
@@ -12,29 +10,23 @@ def show
     render_me_resource
   end
 
-  def show_identity
-    render json: @user_identity
-  end
-
-  def get_identity
-    @user_identity = current_user.identity(params[:service])
-    if @user_identity.blank?
-      logger.warn "User '#{current_user.email}' attempted to access an identity which does not exist: #{params[:service]}"
-      render_error "No such identity '#{params[:service]}' currently exists for this user", :not_found
-    end
-  end
-
   def delete_identity
     # We assume that the `service` param has been validated by the route constraints
-    @user_identity.destroy
+    user_identity = current_user.identity(params[:service])
+    if !user_identity.nil?
```

Comment:
Alternatively you can call destroy on `user_identity` without checking for presence and rescue `ActiveRecord::RecordNotFound` exception.